crates/ruff/src/rules/isort/helpers.rs

```diff
@@ -102,103 +102,148 @@ fn match_docstring_end(body: &[Stmt]) -> Option<Location> {
     Some(stmt.end_location.unwrap())
 }
 
-/// Find the end of the first token that isn't a docstring, comment, or
-/// whitespace.
-pub fn find_splice_location(body: &[Stmt], locator: &Locator) -> Location {
-    // Find the first AST node that isn't a docstring.
-    let mut splice = match_docstring_end(body).unwrap_or_default();
+/// Find the location at which a "top-of-file" import should be inserted,
+/// along with a prefix and suffix to use for the insertion.
+///
+/// For example, given the following code:
+///
+/// ```python
+/// """Hello, world!"""
+///
+/// import os
+/// ```
+///
+/// The location returned will be the start of the `import os` statement,
+/// along with a trailing newline suffix.
+pub fn top_of_file_insertion(
+    body: &[Stmt],
+    locator: &Locator,
+    stylist: &Stylist,
+) -> (&'static str, Location, &'static str) {
+    // Skip over any docstrings.
+    let mut location = if let Some(location) = match_docstring_end(body) {
+        let mut tokens = lexer::lex_located(locator.skip(location), Mode::Module, location)
+            .flatten()
+            .peekable();
```

Comment:
What's the reason for using a `peekable` iterator. We only call `tokens.peek()` once and don't use the iterator any further after. 
```suggestion
        let first_token = lexer::lex_located(locator.skip(location), Mode::Module, location)
            .flatten()
            .next();
```